examples/step-68/step-68.cc

```diff
@@ -0,0 +1,810 @@
+    // be created once, so we might as well have set them up in the constructor
+    // of this class, but for the purpose of this example we want to group the
+    // particle related instructions.
+    background_triangulation.signals.cell_weight.connect(
+      [&](
+        const typename parallel::distributed::Triangulation<dim>::cell_iterator
+          &cell,
+        const typename parallel::distributed::Triangulation<dim>::CellStatus
+          status) -> unsigned int { return this->cell_weight(cell, status); });
+
+    background_triangulation.signals.pre_distributed_repartition.connect(
+      std::bind(
+        &Particles::ParticleHandler<dim>::register_store_callback_function,
+        &particle_handler));
+
+    background_triangulation.signals.post_distributed_repartition.connect(
+      std::bind(
+        &Particles::ParticleHandler<dim>::register_load_callback_function,
+        &particle_handler,
+        false));
+
+    // This initializes the background triangulation where the particles are
+    // living and the number of properties of the particles.
+    particle_handler.initialize(background_triangulation, mapping, 1 + dim);
+
+    // We create a particle triangulation which is solely used to generate
+    // the points which will be used to insert the particles. This
+    // triangulation is an hyper_shell which is off-set from the
+    // center of the simulation domain. This will be used to generate a
+    // disk filled with particles which will allow an easy monitoring
+    // of the motion due to the vortex.
+    Point<dim> center;
+    center[0] = 0.5;
+    center[1] = 0.75;
+    if (dim == 3)
+      {
+        center[2] = 0.5;
+      }
+
+    const double outer_radius = 0.15;
+    const double inner_radius = 0.01;
+
+    parallel::distributed::Triangulation<dim> particle_triangulation(
+      MPI_COMM_WORLD);
+
+    GridGenerator::hyper_shell(
+      particle_triangulation, center, inner_radius, outer_radius, 6);
+    particle_triangulation.refine_global(par.particle_insertion_refinement);
+
+    // We generate the necessary bounding boxes for the particles generator
+    // These bounding boxes are required to quickly identify in which
+    // processors and which cell the inserted particle lies.
+    const auto my_bounding_box = GridTools::compute_mesh_predicate_bounding_box(
+      background_triangulation, IteratorFilters::LocallyOwnedCell());
+    const auto global_bounding_boxes =
+      Utilities::MPI::all_gather(MPI_COMM_WORLD, my_bounding_box);
+
+    // The quadrature points particle generator generates particles only
+    // on locally owned active cells. We therefore count how many of those
+    // are present in the triangulation, as this will be required to
+    // initialize the properties.
+    unsigned int n_locally_owned_cells = 0;
+    for (const auto &cell : particle_triangulation.active_cell_iterators())
+      if (cell->is_locally_owned())
+        n_locally_owned_cells++;
```

Comment:
I think that this whole loop is equivalent to writing `particle_triangulation.n_locally_owned_cells()`.